lib/tjbot.js

```diff
@@ -1137,52 +1162,73 @@ TJBot.prototype.speak = function(message) {
+        self.playerOptions.filename = soundFile
+    } else {
+        self.playerOptions = Object.assign({}, self.playerOptions, soundFile);
+    }
+
 
-    return new Promise(function(resolve, reject) {
-        // if we don't have a speaker, throw an error
-        if (self._soundplayer == undefined) {
-            reject(new Error("unable to play audio, TJBot hardware doesn't include a \"speaker\""));
-            return;
-        }
 
-        // initialize soundplayer lib
-        var speakerOptions = {
-            filename: soundFile,
-            gain: 100,
-            debug: true,
-            player: "aplay", // "afplay" "aplay" "mpg123" "mpg321"
-            device: self.configuration.speak.speakerDeviceId
+    if (self.isPlaying) {
+
+        if (self.configuration.speak.queueSpeech || self.playerOptions.queueSpeech) {
+            winston.debug(this.speechQueue.length + " items in Queue. Queing audio file with parameter ", self.playerOptions);
+            winston.verbose("You attempted to play audio while ", self.playerOptions.filename, " file was playing. Queing ");
+            this.speechQueue.push(self.playerOptions)
+        } else {
+            winston.verbose("You attempted to play audio file ", self.playerOptions.filename, " while a file was playing. To enable Queuing, add `speak: {queueSpeech: true}` to tjconfig ");
         }
-        var player = new self._soundplayer(speakerOptions);
+    } else {
+        self.isPlaying = true;
+        // pause listening while we play a sound -- using the internal
+        // method to avoid a capability check (and potential fail if the TJBot
+        // isn't configured to listen)
+        self._pauseListening();
 
-        winston.debug("Playing audio with parameters: ", speakerOptions);
+        return new Promise(function(resolve, reject) {
+            // if we don't have a speaker, throw an error
+            if (self._soundplayer == undefined) {
+                reject(new Error("unable to play audio, TJBot hardware doesn't include a \"speaker\""));
+                return;
+            }
 
-        player.on('complete', function() {
-            winston.debug("audio playback finished");
+            var player = new self._soundplayer(self.playerOptions);
 
-            // resume listening
-            self._resumeListening();
+            winston.debug("Playing audio with parameters: ", self.playerOptions);
 
-            // done
-            resolve();
-        });
+            player.on('complete', function() {
+                winston.debug("audio playback finished");
 
-        player.on('error', function(err) {
-            winston.error('Error occurred while playing audio :', err);
-        });
+                // resume listening
+                self._resumeListening();
+                self.isPlaying = false
 
-        // play the audio
-        player.play(soundFile);
-    });
+                // done
+                resolve();
```

Comment:
@victordibia this is probably the most important comment I have, please read this if you read nothing else  ;)

is the correct behavior to resolve the promise first and then play back anything else in the speech queue, or play back everything in the speech queue before resolving the promise?  i'm wondering about the semantics of code such as:

```
tj.speak("hello").then({
    tj.speak("world");
}).then({
    tj.speak("how are you");
}).then({
    tj.speak("feeling today");
}).then({
    tj.shine('yellow');
})
```

*question*: when does the light shine yellow?  It could be after all of the speech has been uttered (what I would guess from reading this code), or immediately when tj starts speaking "hello", which is what I think will happen based on the introduction of the speech queue.

*harder question*: I tried implementing a speech queue once early on and it didn't go well, precisely because of the difficulties in making an asynchronous behavior (i.e. the sound player) synchronous. the solution was to use promises, which is what our current code does.  thus, the above example would, right now (before accepting this PR), speak the entirety of the message before shining yellow.  given that promises are correctly handling the "queuing" of speech, is there really a need for a separate speech queue?  in other words, if i'm already writing my code as above, will anything ever actually be inserted into the `speechQueue`?  i could see an argument for making `tj.speak()` itself a synchronous method such that the following code would work:

```
tj.speak("hello");
tj.speak("world");
tj.speak("how are you");
tj.speak("feeling today");
```

But this is kind of the whole point of using promises… as far as I know, you couldn't "block" the return of `tj.speak()` until a callback is invoked because it would lock up the entire node process.